booster/tools/booster/Proxy.hs

```diff
@@ -292,18 +292,21 @@ respondEither cfg@ProxyConfig{statsVar, boosterState} booster kore req = case re
                 else "Iterating execute request at " <> show currentDepth
         -- calculate depth limit, considering possible forced Kore simplification
         let mbDepthLimit = case (cfg.forceFallback, r.maxDepth) of
-                (Just (Depth forceDepth), Just (Depth maxDepth)) ->
-                    if cDepth + forceDepth < maxDepth
-                        then Just $ Depth forceDepth
-                        else Just $ Depth $ maxDepth - cDepth
+                (Just (Depth forceDepth), Just (Depth maxDepth))
+                    | cDepth + forceDepth < maxDepth ->
+                        cfg.forceFallback
+                    | otherwise ->
+                        Just $ Depth $ maxDepth - cDepth
                 (Just forceDepth, _) -> Just forceDepth
                 (_, Just maxDepth) -> Just $ maxDepth - currentDepth
                 _ -> Nothing
         (bResult, bTime) <- Stats.timed $ booster (Execute r{maxDepth = mbDepthLimit})
         case bResult of
             Right (Execute boosterResult)
                 -- the execution reached the depth bound due to a forced Kore simplification
-                | boosterResult.reason == DepthBound && isJust cfg.forceFallback -> do
```

Comment:
Once `cDepth` was reaching `maxDepth` (see  above, when `maxDepth < forceDepth`) we were starting to make requests with `max-depth: 0` and looping.